From 0ace941c07947695b9480bf57ba21797c79cd847 Mon Sep 17 00:00:00 2001
From: Sriharsha Govinda Raj <sriharsha.govinda.raj@intel.com>
Date: Thu, 19 May 2016 15:42:46 +0530
Subject: [PATCH 17/78] EM/power_supply_charger:set MAX_THROTTLE_STATE for
 multiple chargers

MAX_THROTTLE_STATE was only set to the first charger which
was regsiter and was ignored for subsiquent chargers.
This patch supports setting MAX_THROTTLE_STATE for multiple
chargers.

Change-Id: I2a30ea670bbc64cbf4b141f6c05957a757dab718
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-32726
Signed-off-by: Sriharsha <sriharsha.govinda.raj@intel.com>
Reviewed-on: https://android.intel.com:443/509244
---
 drivers/power/power_supply_charger.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/power/power_supply_charger.c b/drivers/power/power_supply_charger.c
index 26a9652..38d1c9e 100644
--- a/drivers/power/power_supply_charger.c
+++ b/drivers/power/power_supply_charger.c
@@ -1017,10 +1017,12 @@ int power_supply_register_charger(struct power_supply *psy)
 {
 	int ret = 0;
 
-	if (!psy_chrgr.is_cable_evt_reg && psy) {
-		ret = register_usb_notifier();
+	if (psy) {
+		if (!psy_chrgr.is_cable_evt_reg) {
+			ret = register_usb_notifier();
+			schedule_work(&notifier_work);
+		}
 		SET_MAX_THROTTLE_STATE(psy);
-		schedule_work(&notifier_work);
 	}
 	psy_chrgr.is_cable_evt_reg = true;
 	return ret;
-- 
1.9.1

